src/components/PeopleTable.tsx

```diff
@@ -1,683 +1,282 @@
-export const PeopleTable = () => {
+import classNames from 'classnames';
+import { useCallback, useEffect, useMemo } from 'react';
+import {
+  Link,
+  useLocation,
+  useParams,
+  useSearchParams,
+} from 'react-router-dom';
+import { Person } from '../types';
+import { PersonNav } from './PersonNav';
+import { SearchLink } from './SearchLink';
+
+enum TypeSort {
+  'Name' = 'name',
+  'Sex' = 'sex',
+  'Born' = 'born',
+  'Died' = 'died',
+}
+
+enum SearchParams {
+  'Sex' = 'sex',
+  'Query' = 'query',
+  'Sort' = 'sort',
+  'Order' = 'order',
+  'Centuries' = 'centuries',
+}
+
+type Props = {
+  listPeople: Person[] | [],
+  setNotFound: (empty: boolean) => void
+};
+
+export const PeopleTable: React.FC<Props> = ({
+  listPeople,
+  setNotFound,
+}) => {
+  const [searchParams] = useSearchParams();
+  const { slug = '' } = useParams();
+  const { search } = useLocation();
+  const sex = searchParams.get(SearchParams.Sex);
+  const query = searchParams.get(SearchParams.Query);
+  const centuries = searchParams.getAll(SearchParams.Centuries);
+
+  const sortBy = useCallback((type: string) => {
+    return {
+      order: searchParams.get(SearchParams.Sort) === type
+        && !searchParams.get(SearchParams.Order)
+        ? 'desc' : null,
+      sort: searchParams.get(SearchParams.Sort) !== type
+        || (searchParams.get(SearchParams.Sort) === type
+          && !searchParams.get(SearchParams.Order))
+        ? type : null,
+    };
+  }, [searchParams]);
+
+  const renderArrows = (type: string) => {
+    return {
+      'fa-sort-up': searchParams.get(SearchParams.Sort) === type
+        && !searchParams.get(SearchParams.Order),
+      'fa-sort-down': searchParams.get(SearchParams.Order)
+        && searchParams.get(SearchParams.Sort) === type,
+    };
+  };
+
+  const filterBy = (array: Person[],
+    filter: {
+      sex: string | null,
+      query: string | null,
+      centuries: string[] | null
+    }) => {
+    let copy = [...array];
+
+    if (filter.sex) {
+      copy = copy.filter((person: Person) => person.sex === filter.sex);
+    }
+
+    if (filter.query) {
+      copy = copy.filter((person: Person) => {
+        return person.name.toLowerCase()
+          ?.includes(filter.query?.toLowerCase() as string)
```

Comment:
You can move `filter.query?.toLowerCase()` to variable to make it simpler, you can also move this function to helper